group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1560,13 +1561,15 @@ private List<Record> consumerGroupStaticMemberGroupLeave(
      * @param records   The list of records to be applied to the state.
      * @return The append future to be applied.
      */
-    private CompletableFuture<Void> consumerGroupFenceMember(
+    private <T> CoordinatorResult<T, Record> consumerGroupFenceMember(
         ConsumerGroup group,
         ConsumerGroupMember member,
-        List<Record> records
+        List<Record> records,
+        T response
     ) {
         if (validateOnlineDowngrade(group, member.memberId())) {
-            return convertToClassicGroup(group, member.memberId(), records);
+            CompletableFuture<Void> appendFuture = convertToClassicGroup(group, member.memberId(), records);
```

Comment:
`convertToClassicGroup` is only used by this method, would it make sense to move `CoordinatorResult` to it?

Reply:
moved `CoordinatorResult` to `convertToClassicGroup`.